governance/assessments/projects/karmada/2023-08-29.md

```diff
@@ -0,0 +1,140 @@
+
+* Complete - the content of the governance documentation is fully detailed and does not leave any question to the reader.
+* Partial - the content of the governance documentation is missing some information and would leave the reader with questions or some level of misunderstanding.
+* Missing - the documentation is absent, wholly undiscoverable, or woefully inadequate in meeting the objectives of that governance content. The reader cannot act on the content that is available.
+* Unknown - status cannot be assessed at this time
+
+| Governance Area | Coverage | Documents | Finding Notes |
+|:----------------|:--------:|:------:|:--------------|
+| Project Purpose | Complete | [Docs](https://karmada.io/docs/) and [README](https://github.com/karmada-io/karmada/tree/master#why-karmada)| |
+| Maintainer List | Complete | [MAINTAINERS.md](https://github.com/karmada-io/karmada/blob/master/MAINTAINERS.md) | It has been the same 7 people since the file was created |
+| Code of Conduct | Partial | [Main repo](https://github.com/karmada-io/karmada/blob/master/CODE_OF_CONDUCT.md) and [Community repo](https://github.com/karmada-io/community/blob/main/CODE_OF_CONDUCT.md) | Karmada currently has two CoCs, it would be helpful to clarifiy which they are using |
+| Contributor Guide | Complete | [CONTRIBUTING.md](https://github.com/karmada-io/karmada/blob/master/CONTRIBUTING.md) | Could be helpful to flesh this out and make it more project specific |
+| Contributor Ladder | Partial | [Community Membership](https://github.com/karmada-io/community/blob/main/community-membership.md) | The roles are laid out, but the owners files are not in synce with the approvers and reviewers |
+| Maintainer Lifecycle | Partial | [Community Membership](https://github.com/karmada-io/community/blob/main/community-membership.md#maintainer) | Adding emeritus process when revoking would be helpful |
+| Decision-making | Complete | [Voting](https://github.com/karmada-io/community/blob/main/GOVERNANCE.md#voting) | Mainly based on lazy consensus |
+| Code and Docs Ownership | Partial | [Owners](https://github.com/karmada-io/community/blob/main/OWNERS), [Owners](https://github.com/karmada-io/karmada/blob/master/OWNERS), [Approvers](https://github.com/karmada-io/community/blob/main/APPROVERS.md), and [Reviewers](https://github.com/karmada-io/community/blob/main/REVIEWERS.md) | All of the files are currently out of sync with each other. It would be helpful to have a single source of truth |
+| Security Reporting and response | Complete | [Security.md ](https://github.com/karmada-io/community/blob/main/security-team/SECURITY.md)| |
+| Communication and Meetings | Complete | [Meetings](https://github.com/karmada-io/karmada/blob/master/README.md#meeting) | This would be helpful to link or reference in the contributing guide |
+
+#### Sub-projects, plugins, and related
+
+The project includes the following sub-projects, plugins, and other notable divisions:
+
+None
```

Comment:
In the karmada-io github organization there is a fork of the kubernetes nginx-ingress with multi-cluster support:
https://github.com/karmada-io/multi-cluster-ingress-nginx

This project is referenced in https://karmada.io/docs/userguide/service/multi-cluster-ingress/. Is this only a sample implementation, some added functionality to be later added to nginx-ingress or a supported component by the karmada project? Is it still maintained (last commit was a while ago)?

Also in the topic of plugins/sub-projects, the project supports providers for its APIs:
https://github.com/karmada-io/multicluster-cloud-provider

Is there info on where they're hosted and who maintains them?

Reply:
>  Is this only a sample implementation, some added functionality to be later added to nginx-ingress or a supported component by the karmada project? Is it still maintained (last commit was a while ago)?

Yes, it's a sample implementation. Thanks for the reminder, I think we can document it somewhere, probably at the repo README as well as the website.  Does that make sense?

The `multicluster-cloud-provider`, is maintained by the Karmada community and consists of public interfaces as well as some common implementations.  The users would implement their cloud-specified providers based on that.